### DISCUSSION TITLE
Randomize the order of buttons in html-button-response

### DISCUSSION THREAD

**Discussion by synapticlee** [jspsych/jsPsych#3018] [Q&A] [answered]
Hi, I'm using jspsych-6.3.1. I was wondering in html-button-response how one would randomize the order of buttons on a trial-by-trial basis (so the same 'answer' isn't always on the left). I used a javascript shuffle function to literally shuffle the list, but the issue is that 'response' is only mapped to the order of the buttons, and therefore I don't actually know which 'answer' was picked. Is there a way within jspsych to randomize the order of buttons while still knowing which button was picked? 

(At the moment, I'm using html to make the button images, but that's incidental to this question)

**Comment by jodeleeuw** [comment, marked as answer]
Hi @synapticlee ,

You can use the `save_trial_parameters` parameter to record the randomized order of the buttons, and then map the response to button after the fact. The [example in the documentation](https://www.jspsych.org/7.3/overview/plugins/#the-save_trial_parameters-parameter) shows a very similar scenario.

**Reply by synapticlee** [reply]
Oh great, thank you! This works! I think actually because I was shuffling at the timeline block level rather than at the trial level, once I did this, it worked

`choices: function() {
          return jsPsych.randomization.shuffle(jsPsych.timelineVariable('choices'));
        },`
